script.lua

```diff
@@ -5880,6 +5861,11 @@ function setupPlayerArea(params)
                 end
             end
         end
+        costs = modifyCost({color = getTableColor(params.obj), costs = costs})
```

Comment:
is `color` not accessible in here?

Reply:
There is a value of `color` (from the outer `setupPlayerArea()` function) available here, but it'll be out of date if the player has changed colours since. `countItems()` works intentionally only with `selected`, and every part of `selected` is invariant to colour swaps. But the original value of `color` used to retrieve `selected` from `selectedColors` may well be outdated. `params.obj` (the player table) is similarly invariant through colour swaps, so I use that to get the latest value of the colour it corresponds to.

Reply:
Just wrote a quick benchmark. Popped this in the global script, and ran it within TTS immediately after loading the mod, with `\execute Global.call("benchmark")`. Turns out that using the GUID as the table key is actually faster (about 0.5 seconds to 0.8 seconds on my computer). Printing the sum's just to check they both give the same answer, and they do.
```lua
function benchmark()
    local startTime
    local endTime
    local n = 200
    
    startTime = os.clock()
    for it = 1,n do
        local t = {}
        for i,obj in ipairs(getObjects()) do
            t[obj.guid] = i
        end
        for k,v in pairs(t) do
            if getObjectFromGUID(k).hasTag("Fast") then
                t[k] = v - 1
            end
        end
        local sum = 0
        for _,v in pairs(t) do
            sum = sum + v
        end
        if it == 1 then
            log(sum)
        end
    end
    endTime = os.clock()
    
    log("GUID as key: "..(endTime - startTime))
    
    startTime = os.clock()
    for it = 1,n do
        local t = {}
        for i,obj in ipairs(getObjects()) do
            t[obj] = i
        end
        for k,v in pairs(t) do
            if k.hasTag("Fast") then
                t[k] = v - 1
            end
        end
        local sum = 0
        for _,v in pairs(t) do
            sum = sum + v
        end
        if it == 1 then
            log(sum)
        end
    end
    endTime = os.clock()
    
    log("Reference as key: "..(endTime - startTime))
end
```